### PR TITLE
technology decisions

### PR DESCRIPTION
**Pull request by dwhiteddsoft**
This PR contains three technology decision documents containing reasons we made them. These technology decisions were in regards to:
- device language choice
- AKS vs Functions
- container management on the device

### CODE REVIEW

File: docs/tech-decisions/device-programming-language.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+  - [Notes](#notes)
+
+## Summary
+
+### Issue
+
+We need to choose a programming languages for the software that runs on the embedded device under the following constraints:
+
+- All processes must run on a constrained embedded processor
+- All processes must run under 256Mb of memory
```
Comment (rkrithivasan) [inline docs/tech-decisions/device-programming-language.md:29]: Is it 256 Mb or MB, typically size is in MB (bytes).

**Review by rkrithivasan — APPROVED** [review]

File: docs/tech-decisions/aks-vs-functions.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    - [Related decisions](#related-decisions)
+    - [Related requirements](#related-requirements)
+    - [Related principles](#related-principles)
+  - [Notes](#notes)
+
+## Summary
+
+### Issue
+
+What is the right choice for a compute environment (using containers) in the cloud given cloud scale requirements on the high end for Bosch? The choices on the table would be:
```
[the remark this answers is not in the document]
Comment (dwhiteddsoft) [reply docs/tech-decisions/aks-vs-functions.md]: updated per comments

File: docs/tech-decisions/aks-vs-functions.md [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+The implementation team has vast familiarity with both compute environments. From this perspective, we simply choose what compute stack gives us the most flexibility and cloud scale needed.
+
+### Argument
+
+Every service in Azure has limits. All service limits can be changed via a support ticket being opened with Azure support. When you compare the [App Service limits](https://docs.microsoft.com/en-us/azure/azure-resource-manager/management/azure-subscription-service-limits#app-service-limits) with [Azure Kubernetes limits](https://docs.microsoft.com/en-us/azure/azure-resource-manager/management/azure-subscription-service-limits#azure-kubernetes-service-limits), we have a longer "run way" to start hitting limits and more flexibility within.
+
+### Implications
+
+Deploying our containers to AKS as opposed to App Service Environments is a degree harder in general. As Azure App Service is a PaaS based service, deployment and management are very easy. And while AKS is also a PaaS based service, the service level it provides to the user is at a much lower level than the App Service.
```
[the remark this answers is not in the document]
Comment (dwhiteddsoft) [reply docs/tech-decisions/aks-vs-functions.md]: updated per comments

File: docs/tech-decisions/podman-vs-runc-vs-crun.md [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+## Summary
+
+### Issue
+
+We will endeavor to build a container management solution on a very constrained device as part of this project. Many of the container management/orchestration engines that exist simply will not be practical to run on a constrained embedded device (Docker, containerd, etc.). Options become much more limited when looking at deployment on a constrained embedded devices. Our range of choices in this case get much more narrow and look like a choice between wrapping [podman](https://podman.io/) or going a lower level and wrapping [runc](https://github.com/opencontainers/runc) or [crun](https://github.com/containers/crun).
+  
+### Decision
+
+We will be using podman for container management and orchestration. The reasons behind this are:
```
[the remark this answers is not in the document]
Comment (dwhiteddsoft) [reply docs/tech-decisions/podman-vs-runc-vs-crun.md:30]: I can find no requirements for podman other than Linux (I imagine POSIX base)

**Pull request merged**